docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,21 +1,25 @@
 [//]: # (title: Hierarchical project structure)
 
-Multiplatform projects have support for hierarchical structure.
-This means you can form a hierarchy of source sets for sharing the common code among some, but not all,
-[supported targets](multiplatform-dsl-reference.md#targets).
-
-The Kotlin toolchain provides the correct default dependencies and locates the API surface area available in the shared code.
-This prevents cases like the use of a macOS-specific function in code shared for Windows. It opens up a variety of opportunities:
-
-* Using platform-dependent libraries in source sets shared among several native targets. One popular case is having access
-  to iOS-specific dependencies like `Foundation` when sharing code across all iOS targets.
-* Sharing code in similar targets, as well as publishing and consuming libraries with granular APIs targeting similar platforms.
+Multiplatform projects support hierarchical structures.
+This means you can form a hierarchy of intermediate source sets for sharing the common code among some, but not all,
+[supported targets](multiplatform-dsl-reference.md#targets). Introducing intermediate source sets has some important advantages:
+
+* If you're a library author, you might want to provide a specialized API for some, but not all targets. For example,
+  an intermediate source set for Kotlin/Native targets, but not for Kotlin/JVM may come in handy.
+* You might want to use platform-dependent libraries in your project. Introducing an intermediate source set allows
+  you to utilize that specialized API among several native targets. For example,
+  having access to iOS-specific dependencies like Foundation when sharing code across all iOS targets.
+* Some libraries aren't available for some platforms. Specifically, native libraries are only available for source sets
+  that compile to Kotlin/Native. Intermediate source set will solve this issue.
+* The Kotlin toolchain ensures that each source set has access only to the API that is available for all targets to which
```

Comment:
Very minor: I think this point isn't the _reason_ why you want to have an intermediate source set, it's just a nice feature we provide to you **if** you had to use an intemdetiate source set (by any of the reasons above). So I'd just remove it from the bullet point list